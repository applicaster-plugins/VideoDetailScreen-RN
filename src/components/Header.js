import React, { Component } from 'react';
import { navigation } from 'react-native-zapp-bridge';
import { View, Image, TouchableHighlight } from 'react-native';
import assets from '../utils/assetsMap';
import { styles } from './Header.styles';

export default class Header extends Component {
  constructor(props) {
    super(props);
    this.styles = styles();
  }

  handleClose() {
    navigation.closeModalScreen();
  }

  handleShare() {
    console.log('share');
  }

  handleCast() {
    console.log('cast');
  }

  render() {
    return (
      <View style={this.styles.container}>
        <TouchableHighlight onPress={this.handleClose} style={this.styles.btn}>
          <Image source={assets.iconClose} style={this.styles.closeImg} />
        </TouchableHighlight>
      </View>
    );
  }
}
