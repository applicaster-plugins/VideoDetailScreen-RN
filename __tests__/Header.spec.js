import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../src/components/Header';

jest.mock('../src/utils/zappConfig');

it('Header renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();

  expect(tree).toMatchSnapshot();
});
