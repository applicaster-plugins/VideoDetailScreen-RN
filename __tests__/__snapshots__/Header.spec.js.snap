// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(16, 17, 22)",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 5,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "RN_Video_Detail_Video_Close_Btn",
        }
      }
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
  </View>
</View>
`;
